sips/reduce_signatures_fork.md

```diff
@@ -0,0 +1,163 @@
+}
+```
+
+### Decided messages
+
+With the proposed change, a *decided* message is identified by a `SignedSSVMessage` message with multiple signers and signatures.
+
+**Creation**: To create a *decided* message, no BLS aggregation is performed anymore. Now, the commit message data is sent under `SignedSSVMessage.Data` and the quorum (or more) of signers are listed in `SignedSSVMessage.OperatorID` along with their respective signatures in `SignedSSVMessage.Signature`.
+
+**Verification**: To verify a decided message, no more BLS verification is performed. Now, the number of signatures in `SignedSSVMessage.Signature` should be greater or equal to the cluster's quorum and the n-th signature should be correct for the public key of the n-th signer of `SignedSSVMessage.OperatorID` for $n = 0, 1, ..., len(Signatures) - 1$.
+
+
+### Consensus message processing
+
+The consensus module currently accepts `SignedMessage` messages. Now, it must accept `SignedSSVMessage` messages. `SignedSSVMessage` is required instead of the simple `Message` in order to allow the creation of messages with justifications.
+
+Except by messages inside justifications, the consensus module should not verify the signature of the received `SignedMessage` since it assumes that the message validation module already performed this check.
+
+Every sub-processing function of the consensus module, such as the `UponProposal`, `UponPrepare`, etc, should also receive a `SignedSSVMessage` instead of the deprecated `SignedMessage`.
+
+The message container must also stop storing `SignedMessage` objects in favor of storing `SignedSSVMessage` objects.
+
+### Partial signature message processing
+
+Currently, the partial signature processing functions accept a `SignedPartialSignatureMessage`. Now, it should accept a `SignedSSVMessage`. Similarly to the consensus module, it should assume that the message validation module already verified the RSA signature.
+
+### Validator
+
+Currently, the `Validator` module receives an `SSVMessage` object as the input to its `ProcessMessage` function. Now, it should receive a `SignedSSVMessage`. Moreover, it should pass the same `SignedSSVMessage` object to the consensus and partial signature processing functions (and not one of its child types).
+
+### Runner
+
+Aligned to the changes on the `Validator` module, the `Runner` module should accept only `SignedSSVMessage` messages to its processing functions.
+
+### Message Validation
+
+Message validation should stop decoding the `SignedSSVMessage` to the deprecated `SignedMessage` or `SignedPartialSignatureMessage`. To check the content of the message it must decode the message data directly to one of the `Message` or `PartialSignatureMessages` types.
+
+Message validation should only allow multiple signers in the `SignedSSVMessage` message for *decided* messages (a check that was previously done for the `SignedMessage` type). It should verify it according to the [description above](#decided-messages).
```

Comment:
knowledge base and rules should be updated after this is merged.

Reply:
Yes, it will. Thanks!